apitools/base/py/base_api.py

```diff
@@ -454,8 +457,11 @@ def DeserializeMessage(self, response_type, data):
     def FinalizeTransferUrl(self, url):
         """Modify the url for a given transfer, based on auth and version."""
         url_builder = _UrlBuilder.FromUrl(url)
-        if self.global_params.key:
+        if getattr(self.global_params, 'key', None):
```

Comment:
Any reason for changing this?

Reply:
Not ideal but because tests were failing.
I spent a while trying to figure out why, and it was much easier to change the code to this equivalent line.
Also, this makes `FinalizeTransferUrl` a little more robust. If `self.global_params` doesn't have `key`, like in the tests, the method won't fail and will still be able to persist custom endpoints.